flink-table/flink-table-common/src/main/java/org/apache/flink/table/sources/LimitableTableSource.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.sources;
+
+import org.apache.flink.annotation.Experimental;
+
+/**
+ * Adds support for limiting push-down to a {@link TableSource}.
+ * A {@link TableSource} extending this interface is able to limit the number of records.
+ *
+ * <p>After pushing down, source only needs to try its best to limit the number of output records,
+ * but does not need to guarantee that the number must be less than or equal to the limit.
```

Comment:
Could you add more description here about split limit and planner guarantee (they are mentioned in the rule javadoc)? I think these are useful to users.

Reply:
I think this word just tell user: there is no limit, you just do what you can do.
The comments in `PushLimitIntoTableSourceScanRule` is difficult to understand without code, I think we can just add a link to `PushLimitIntoTableSourceScanRule`.

Reply:
Btw, `less than or equal to the limit`, the source shouldn't generate records less than the limit, right?

Reply:
maybe, if the total records are less than the limit, it could be.

Reply:
OK, that meakes sense.